lib/src/conflicts.rs

```diff
@@ -278,27 +278,39 @@ pub fn parse_conflict(input: &[u8], num_removes: usize, num_adds: usize) -> Opti
 }
 
 fn parse_conflict_hunk(input: &[u8]) -> MergeHunk {
+    let mut diff_seen = false;
     let mut minus_seen = false;
     let mut plus_seen = false;
     let mut body_seen = false;
     let mut removes = vec![];
     let mut adds = vec![];
     for line in input.split_inclusive(|b| *b == b'\n') {
-        if line == CONFLICT_MINUS_LINE {
+        if line == CONFLICT_DIFF_LINE {
+            diff_seen = true;
+            if body_seen {
+                minus_seen = false;
+                plus_seen = false;
+                body_seen = false;
+            }
+            removes.push(vec![]);
+            adds.push(vec![]);
+        } else if line == CONFLICT_MINUS_LINE {
             minus_seen = true;
             if body_seen {
+                diff_seen = false;
                 plus_seen = false;
                 body_seen = false;
             }
             removes.push(vec![]);
         } else if line == CONFLICT_PLUS_LINE {
             plus_seen = true;
             if body_seen {
+                diff_seen = false;
                 minus_seen = false;
                 body_seen = false;
             }
             adds.push(vec![]);
-        } else if minus_seen && plus_seen {
```

Comment:
Do we need to support the old -------/+++++++ marker?

The current version doesn't, and the seen flags can be cleaned up by a subsequent patch.

Reply:
I don't think we need to support the old format. It's easy enough to recover.